R/utils_files.R

```diff
@@ -67,3 +71,16 @@ file_copy <- function(from, to) {
     file.copy(from = from, to = to, overwrite = TRUE)
   }
 }
+
+trust_timestamps <- function(path) {
+  out <- rep(FALSE, length(path))
+  exists <- file.exists(path)
+  safe <- c(
```

Comment:
I would have thought reasonably high precision timestamps were now the default rather than the exception so you could maybe make this an unsafe list rather than a safe list? But many there just aren't that many file systems in common usage today and it doesn't matter either way.

Reply:
Otherwise chatGPT suggests that you might be able to add NTFS, XFS, btrfs, and ZFS here. But @gaborcsardi is likely to have a better intuition than me.

Reply:
AFAIR NTFS resolution also depends on the API you use to query it. THe rest are probably good, their market share is also really-really small, so it does not really matter.

Reply:
> I would have thought reasonably high precision timestamps were now the default rather than the exception so you could maybe make this an unsafe list rather than a safe list? 

Good point, I will switch to an unsafe list.

Reply:
da90f3ebc734c3a37b7fe29d12530415de8def11 adds an unsafe list. I included the file systems from https://en.wikipedia.org/wiki/List_of_file_systems whose respective Wikipedia articles list "date resolution" entries worse than 1 millisecond.